components/MealPlan.js

```diff
@@ -4,16 +4,54 @@ import {
   Text,
   StyleSheet,
 } from "react-native";
+import { connect } from "react-redux";
 
-
+const mapStateToProps = (state) => {
+  return { username: state.user.username, password: state.user.password};
+};
 import Colors from "../constants/Colors";
-export default class MealPlan extends Component {
+import user from "../data/redux/user";
+export class MealPlan extends Component {
 
     state = {
-        tams:1,
-        flex:1,
+        tams:"",
+        flex:"",
     };
 
+    constructor(props){
+      super(props);
+    }
+
+    componentDidMount() {
+      try {
+        let formdata = new FormData();
+
+        formdata.append("username", this.props.username);
+        formdata.append("password", this.props.password);
+  
+        fetch("http://miranda.caslab.queensu.ca/GetMealPlan",
+        {
+          method: "post",
+          headers: {
+            "Content-Type": "multipart/form-data",
+          },
+          body: formdata,
+        }
+      )
+          .then((response) => response.json())
+          .then((json) => {
+            this.setState({ tams: json.TAMS });
+            this.setState({ flex: json.FLEX });
```

Comment:
merge into one single setState. This will refresh the component twice. U can say {tams: json.Tams, flex: json.FLEX}